vpn/tunnel.go

```diff
@@ -170,6 +243,40 @@ func sinkEntryToPb(e slog.SinkEntry) *Log {
 	return l
 }
 
+func convertWorkspaceUpdate(update *proto.WorkspaceUpdate) *PeerUpdate {
+	out := &PeerUpdate{
+		UpsertedWorkspaces: make([]*Workspace, len(update.UpsertedWorkspaces)),
+		UpsertedAgents:     make([]*Agent, len(update.UpsertedAgents)),
+		DeletedWorkspaces:  make([]*Workspace, len(update.DeletedWorkspaces)),
+		DeletedAgents:      make([]*Agent, len(update.DeletedAgents)),
+	}
+	for i, ws := range update.UpsertedWorkspaces {
+		out.UpsertedWorkspaces[i] = &Workspace{
+			Id:   ws.Id,
+			Name: ws.Name,
+		}
+	}
+	for i, agent := range update.UpsertedAgents {
+		out.UpsertedAgents[i] = &Agent{
+			Id:   agent.Id,
+			Name: agent.Name,
```

Comment:
missing workspace_id, fqdn.

Also missing ip address and last handshake, but we can save those for another PR.